Modules/itertoolsmodule.c

```diff
@@ -2712,33 +2723,55 @@ combinations_next(combinationsobject *co)
          */
         assert(r == 0 || Py_REFCNT(result) == 1);
 
+        /* Seems like we allow 0-length combination ranges. */
```

Comment:
Since `comb(n, 0) == 1`, yes, whenever `r == 0` we produce one result (an empty tuple). So don't say "seems like" 😉 I think it would be better to move `if (0 == r) goto empty;` to the top of this `else` block, and get rid of that special case at once. We already returned an empty tuple the first time this function was called. The only thing to do on the second call in the `r == 0` case is `goto empty`. All the code before this line in the `else` block is useless churn when `r == 0`. Very rarely executed, but useless all the same,  so it would be clearer to skip it. Then the `assert` can lose the `r == 0` clause too.